### PR TITLE
Add experiment name

### LINKED ISSUE #6386: Introduce name for experiments
Currently only ensembles have names, however we should also introduce names for experiments.

### PR DESCRIPTION
**Pull request by aronhoyer**
**Issue**
Resolves #6386

**Approach**
Adds a text input field whose placeholder text is the simulation mode.

If no name is provided, the placeholder text is used as the experiment name.

Screenshot 2023-11-27 at 13 14 24

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dafeda — APPROVED** [review]
LGTM, but good i @oyvindeide takes a look as well.

File: src/ert/storage/local_storage.py [hunk 1]
```diff
@@ -212,6 +212,13 @@ def __init__(
 
                     observations.migrate(self.path)
                     self._add_migration_information(3, "observations")
+                elif version == 4:
+                    from ert.storage.migration import (
+                        experiment_id,
+                    )
+
+                    experiment_id.migrate(self.path)
+                    self._add_migration_information(4, "experiment_id")
```
Comment (oyvindeide) [inline src/ert/storage/local_storage.py:221]: This will have to be added to the other versions as well. The reason is because we always jump straight to the latest version, and dont upgrade iteratively. It might be time to start doing that, but that is outside the scope here.

File: src/ert/storage/local_storage.py [hunk 2]
```diff
@@ -252,17 +259,20 @@ def create_experiment(
         parameters: Optional[List[ParameterConfig]] = None,
         responses: Optional[List[ResponseConfig]] = None,
         observations: Optional[Dict[str, xr.Dataset]] = None,
+        name: Optional[str] = None,
```
Comment (oyvindeide) [inline src/ert/storage/local_storage.py:262]: Should this be defaulted? In the migration it is added explicitly, instead of being None, indicating that maybe it is required

**Comment by oyvindeide** [comment]
Looks like a good improvement! Just a comment about the migrations that needs to be addressed. Just remember to squash commits before merging.

**Review by oyvindeide — APPROVED** [review]
